OpenSim/Actuators/VandenBogert2011Muscle.cpp

```diff
@@ -0,0 +1,1234 @@
+
+    //W (dimensionless) width parameter of the force-length relationship of
+    //      the muscle fiber
+    double fl_width = getFlWidth();
+
+    //AHill (dimensionless) Hill parameter of the force-velocity relationship
+    double fv_AHill= getFvAHill();   //Hill parameter of the f-v relationship
+
+    //FV_{max} (dimensionless) maximal eccentric force
+    double fv_maxMultiplier = getFvmaxMultiplier();
+    //Maximal eccentric force multiplier
+
+    // L_{opt}  (m) Optimal Length of Contractile Element;
+    double optFiberLength = getOptimalFiberLength();
+    //Optimal Length of Contractile Element
+
+    //b (s/m) damping coefficient of damper parallel to the fiber
+    // (normalized to maxIsometricForce)
+    double dampingCoeff = getDampingCoefficient();
+
+    //L_{slack,fiber} Slack length of the parallel elastic element, divided by
+    //          Lceopt
+    double fiberSlackLengthNorm= getNormFiberSlackLength();
+
+    //L_{slack,tendon} (m) slack length of the tendon
+    double tendonSlackLength = getTendonSlackLength();
+
+    //T_{act} (s) Activation time
+    double activTimeConstant = getActivTimeConstant();
+
+    //T_{deact} (s) Deactivation time
+    double deactivTimeConstant = getDeactivTimeConstant();
+
+    //phi_{opt} pennation at Optimal Fiber Length
+    double pennAtOptFiberLength = getPennAtOptFiberLength();
+
+
+    // constants derived from the muscle parameters
+    double vMaxNorm = 10 * optFiberLength;
+    //Maximum shortening velocity in optimal fiber lengths per sec
+
+    // Jacobian Matrices
+    SimTK::Mat22 df_dy;
+    SimTK::Mat22 df_dydot;
+
+
+    //-------Convert projFiberLength & Velocity to fiberLength & Velocity------/
+    double cosPenn;
+    double dcosPenn_dprojFibLen;
+    double fiberLengthNorm;
+    double dfiberLength_dprojFibLen;
+    if (pennAtOptFiberLength<0.01)  {
+        // If pennation is zero, we can't do this because volume is zero,
+        //      and fiberLength ~ projFiberLength
+        cosPenn=1.0;
+        fiberLengthNorm=projFibLenNorm;
+        dfiberLength_dprojFibLen=1;
+        dcosPenn_dprojFibLen=0;
+    }
+    else {
+        double b=sin(pennAtOptFiberLength);
+        fiberLengthNorm=sqrt(pow(projFibLenNorm,2) + pow(b,2));
+        cosPenn = projFibLenNorm/fiberLengthNorm;
+        dfiberLength_dprojFibLen=cosPenn;
+        dcosPenn_dprojFibLen=pow(b,2) / pow(fiberLengthNorm,3);
+
+    }
+
+    // Compute fiberVelocity and its derivatives wrt projFibLen and projFibVel
+    double fiberVelocityNorm = projFibVelNorm*cosPenn;
+    double dfiberVelocityNorm_dprojFibVelNorm = cosPenn;
+    double dfiberVelocityNorm_dprojFibLenNorm =
+            projFibVelNorm * dcosPenn_dprojFibLen;
+
+
+    //---F1 is the normalized isometric force-length relationship at maximum
+    //                                               activation--------------//
+    double fiberExp = (fiberLengthNorm - 1.0) / fl_width;   // [dimensionless]
```

Comment:
Yes - I verified that they are the same.  Should I rename the parameter to match?